pkg/agent/handler/credential_test.go

```diff
@@ -138,7 +138,7 @@ func TestCredentialHandler(t *testing.T) {
 				subError:       tc.subError,
 			}
 
-			handler := NewCredentialHandler(p, c)
+			handler := NewCredentialHandler(p, c, nil)
```

Comment:
Can you please add a test with the oauth provider registry?

Reply:
Yes, still need to add few more tests